### PR TITLE
examples: improve tx_context handling and extend examples

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: examples/example_host.cpp [hunk 1]
```diff
@@ -25,8 +25,12 @@ struct account
 class ExampleHost : public evmc::Host
 {
     std::map<evmc::address, account> accounts;
+    evmc_tx_context tx_context = evmc_tx_context{};
 
 public:
+    ExampleHost() = default;
+    ExampleHost(evmc_tx_context& _tx_context) : tx_context(_tx_context){};
```
Comment (chfast) [inline examples/example_host.cpp]:
```suggestion
    explicit ExampleHost(evmc_tx_context& _tx_context) noexcept : tx_context{_tx_context}{};
```

File: examples/example_host.cpp [hunk 2]
```diff
@@ -25,8 +25,12 @@ struct account
 class ExampleHost : public evmc::Host
 {
     std::map<evmc::address, account> accounts;
+    evmc_tx_context tx_context = evmc_tx_context{};
```
Comment (chfast) [inline examples/example_host.cpp]:
```suggestion
    evmc_tx_context tx_context{};
```

File: examples/example_host.cpp (hunk 1 above)
Comment (axic) [reply examples/example_host.cpp]: Fixed.

File: examples/example_host.cpp [hunk 3]
```diff
@@ -25,8 +25,12 @@ struct account
 class ExampleHost : public evmc::Host
 {
     std::map<evmc::address, account> accounts;
+    evmc_tx_context tx_context{};
 
 public:
+    explicit ExampleHost() = default;
```
Comment (chfast) [inline examples/example_host.cpp]: The `explicit` is only needed for constructors with single argument.

Comment (axic) [reply examples/example_host.cpp]: Fixed.

**Pull request merged**